controllers/machinehealthcheck_controller.go

```diff
@@ -250,7 +250,8 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, logger log
 	// mark for remediation
 	errList := []error{}
 	for _, t := range unhealthy {
-		logger.V(3).Info("Target meets unhealthy criteria, triggers remediation", "target", t.string())
+		condition := conditions.Get(t.Machine, clusterv1.MachineHealthCheckSuccededCondition)
+		logger.Info("Target has failed health check, marking for remediation", "target", t.string(), "reason", condition.Reason, "message", condition.Message)
```

Comment:
Not directly related to this PR, but I'm wondering if it would be good to add helper methods to the Condition type to "stringify" a condition for purposes such as this.

Reply:
Yeah, I looked for this and was surprised we didn't already have one. `String()` on the condition and maybe a `conditions.String(c)` method?

Reply:
Yeah, that is along the lines of what I was thinking, but not sure if we can completely generalize it or should have a few different helpers for different purposes (failure vs success, brief vs verbose, etc). It might have the potential to be a bit of a rabbit hole, so likely best handled as a separate issue.

Reply:
Thanks @benmoss. Is there an example msg when this happened?